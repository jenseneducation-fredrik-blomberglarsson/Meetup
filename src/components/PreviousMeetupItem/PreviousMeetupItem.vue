<template>
  <div class="previousMeetupItem">
    <div class="previousMeetupItemContainer">
      <div class="previousMeetupCard">
        <div class="headerContainer">
          <h3 class="previousMeetupCategory">{{previousMeetup.category}}</h3>
          <h2 class="previousMeetupTitle">{{previousMeetup.title}}</h2>
        </div>
        <div class="content">
          <p class="previousMeetupDesc">{{previousMeetup.desc}}</p>
          <p class="previousMeetupDate">{{previousMeetup.date}}</p>
          <p class="previousMeetupLocation">{{previousMeetup.location}}</p>
          <p class="previousMeetupOrganizer">{{previousMeetup.organizer}}</p>
          <div class="createReview">
            <button
              class="createReviewBtn"
              @click="toggleReview()"
              v-if="!isSubmitted"
            >Create a review!</button>
          </div>
        </div>
      </div>
      <div class="createReviewForm" v-show="showReview">
        <section class="form-group">
          <h3 class="meetupTitle">Submit a review for {{previousMeetup.title}}</h3>
          <input type="text" class="form-control" placeholder="Name..." v-model="reviewData.name" />
        </section>
        <section class="form-group">
          <label for="satisfaction">
            How did you experience this meetup?
            <br />(10 for best, 1 for worst)
          </label>
          <input
            placeholder="1-10"
            min="1"
            max="10"
            type="number"
            id="satisfaction"
            class="form-control"
            v-model.number="reviewData.satisfaction"
          />
        </section>
        <section class="form-group">
          <textarea
            placeholder="Describe your review..."
            id="message"
            rows="5"
            class="form-control"
            v-model="reviewData.message"
          ></textarea>
        </section>
        <button class="submitReview" @click.prevent="submitted(); toggleReview();">Submit!</button>
      </div>
      <div class="displayReview" v-if="isSubmitted">
        <div class="displayReviewBody">
          <p class="reviewName">
            Name:
            <b>{{ reviewData.name }}</b>
          </p>
          <p class="reviewSatisfaction">
            Satisfaction:
            <b>{{ reviewData.satisfaction }}</b>
          </p>
          <p class="reviewMessage" style="white-space: pre">
            Message:
            <b>{{ reviewData.review }}</b>
          </p>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "PreviousMeetupItem",
  props: { previousMeetup: Object },

  // Have to make locally instead of global in store
  data() {
    return {
      showReview: false,
      reviewData: {
        name: "",
        satisfaction: "",
        review: "",
      },
      isSubmitted: false,
    };
  },
  methods: {
    toggleReview() {
      this.showReview = !this.showReview;
    },
    submitted() {
      this.isSubmitted = true;
    },
  },
};
</script>

<style scoped>
.createReviewForm {
  margin-left: 5%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

div {
  display: inline-block;
}
body {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  flex-direction: row;
}

.previousMeetupDate {
  text-decoration: line-through;
}

.previousMeetupItemContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  padding: 30px;
}

.previousMeetupItemContainer .previousMeetupCard {
  max-width: 300px;
  height: 150px;
  background: #ffeb3b;
  margin: 30px 10px;
  padding: 20px 15px;
  display: flex;
  flex-direction: column;
  box-shadow: 0 5px 202px rgba(0, 0, 0, 0.5);
  transition: 0.3s ease-in-out;
}

.previousMeetupItemContainer .previousMeetupCard:hover {
  height: 350px;
  background: linear-gradient(#03a9f4, #f441a5, #ffeb3b, #03a9f4);
  background-size: 400% 400%;
  animation: change 10s ease-in-out infinite;
}

.previousMeetupItemContainer .previousMeetupCard .content {
  margin-top: -140px;
  padding: 10px 15px;
  text-align: center;
  color: #111;
  visibility: hidden;
  opacity: 0;
  transition: 0.3s ease-in-out;
}
.previousMeetupItemContainer .previousMeetupCard:hover .content {
  visibility: visible;
  opacity: 1;
  margin-top: -5px;
  transition-delay: 0.2s;
}
@keyframes change {
  0% {
    background-position: 0 100%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0 100%;
  }
}
</style>