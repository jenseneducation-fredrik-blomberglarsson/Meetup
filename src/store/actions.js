import { getMeetups } from "../data/meetupData";

const actions = {
  async fetchMeetups(context) {
    const data = await getMeetups();
    context.commit("setMeetups", data);
  }
}

export default actions;