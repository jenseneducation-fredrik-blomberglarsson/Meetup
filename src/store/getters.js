const getters = {
  getMeetups: state => {
    return state.meetups;
  },
  getJoinedMeetups: state => {
    return state.joinedMeetups;
  },
  getPreviousMeetups: state => {
    return state.previousMeetups;
  },
  getMeetupToggle: state => {
    return state.meetupToggle;
  },
  getBurgerToggle: state => {
    return state.burgerToggle;
  }
}


export default getters;