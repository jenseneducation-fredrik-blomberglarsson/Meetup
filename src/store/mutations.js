const mutations = {
  setMeetups(state, meetups) {
    state.meetups = meetups;
  }
}

export default mutations;