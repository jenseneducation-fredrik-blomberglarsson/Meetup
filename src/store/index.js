import Vue from 'vue'
import Vuex from 'vuex'
import actions from './actions.js';
import mutations from './mutations.js';
import getters from './getters.js';

Vue.use(Vuex)

//Global state
export default new Vuex.Store({
  state: {
    meetups: [],
    joinedMeetups: [],
    previousMeetups: [],
    meetupToggle: false,
  },

  actions: actions,
  mutations: mutations,
  getters: getters
});