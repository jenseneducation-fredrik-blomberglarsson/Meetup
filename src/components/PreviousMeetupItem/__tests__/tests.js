import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import store from '../../../store/index.js';
import PreviousMeetupItem from '@/components/PreviousMeetupItem/PreviousMeetupItem.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('The page is rendered', () => {
  // Mock previousMeetupItem
  const previousMeetup = {
    id: '01',
    category: 'Film',
    title: 'Movie marathon',
    desc: 'Watch different movies online with film enthusiasts',
    date: 'This meetup has passed',
    location: 'Online',
    organizer: 'Stockholm Film',
  };

  test('Test so that everything renders properly', async () => {
    const wrapper = shallowMount(PreviousMeetupItem, {
      store,
      localVue,
      propsData: {
        previousMeetup
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('Test to check that output is correct on previousMeetup', async () => {
    const wrapper = shallowMount(PreviousMeetupItem, {
      store,
      localVue,
      propsData: {
        previousMeetup
      }
    });

    const previousMeetupCategory = wrapper.find('.previousMeetupCategory');
    const previousMeetupTitle = wrapper.find('.previousMeetupTitle');
    const previousMeetupDesc = wrapper.find('.previousMeetupDesc');
    const previousMeetupDate = wrapper.find('.previousMeetupDate');
    const previousMeetupLocation = wrapper.find('.previousMeetupLocation');
    const previousMeetupOrganizer = wrapper.find('.previousMeetupOrganizer');

    expect(previousMeetupCategory.text()).toContain(previousMeetup.category);
    expect(previousMeetupTitle.text()).toContain(previousMeetup.title);
    expect(previousMeetupDesc.text()).toContain(previousMeetup.desc);
    expect(previousMeetupDate.text()).toContain(previousMeetup.date);
    expect(previousMeetupLocation.text()).toContain(previousMeetup.location);
    expect(previousMeetupOrganizer.text()).toContain(previousMeetup.organizer);
  });

  test('Check that reviewForm does not show when rendered', async () => {
    const wrapper = shallowMount(PreviousMeetupItem, {
      store,
      localVue,
      propsData: {
        previousMeetup
      }
    });
    const reviewForm = wrapper.find('.createReviewForm');
    let elementExists = reviewForm.exists();
    expect(elementExists).toBe(true); // Kollar om elementet existerar

    if (elementExists) {
      let reviewVisible = reviewForm.element.style.display == '!none'; // Kollar om elementet Ã¤r synligt
      expect(reviewVisible).toBe(false);
    }
  })

  test('Check that toggleReview is called when the createReview button is clicked', async () => {
    const wrapper = shallowMount(PreviousMeetupItem, {
      store,
      localVue,
      propsData: {
        previousMeetup
      }
    });

    wrapper.vm.toggleReview = jest.fn();
    const button = wrapper.find('.createReviewBtn');
    await button.trigger('click');
    expect(wrapper.vm.toggleReview).toHaveBeenCalled();
  })

  test('Check that submitted is called when submit button is clicked', async () => {
    const wrapper = shallowMount(PreviousMeetupItem, {
      store,
      localVue,
      propsData: {
        previousMeetup
      }
    });

    wrapper.vm.submitted = jest.fn();
    const button = wrapper.find('.submitReview');
    await button.trigger('click');
    expect(wrapper.vm.submitted).toHaveBeenCalled();
  })
})
