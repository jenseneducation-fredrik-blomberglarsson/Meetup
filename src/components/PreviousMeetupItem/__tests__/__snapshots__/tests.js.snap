// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page is rendered Test so that everything renders properly 1`] = `
<div
  class="previousMeetupItem"
>
  <div
    class="previousMeetupItemContainer"
  >
    <div
      class="previousMeetupCard"
    >
      <div
        class="headerContainer"
      >
        <h3
          class="previousMeetupCategory"
        >
          Film
        </h3>
         
        <h2
          class="previousMeetupTitle"
        >
          Movie marathon
        </h2>
      </div>
       
      <div
        class="content"
      >
        <p
          class="previousMeetupDesc"
        >
          Watch different movies online with film enthusiasts
        </p>
         
        <p
          class="previousMeetupDate"
        >
          This meetup has passed
        </p>
         
        <p
          class="previousMeetupLocation"
        >
          Online
        </p>
         
        <p
          class="previousMeetupOrganizer"
        >
          Stockholm Film
        </p>
         
        <div
          class="createReview"
        >
          <button
            class="createReviewBtn"
          >
            Create a review!
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="createReviewForm"
      style="display: none;"
    >
      <section
        class="form-group"
      >
        <h3
          class="meetupTitle"
        >
          Submit a review for Movie marathon
        </h3>
         
        <input
          class="form-control"
          placeholder="Name..."
          type="text"
        />
      </section>
       
      <section
        class="form-group"
      >
        <label
          for="satisfaction"
        >
          
          How did you experience this meetup?
          
          <br />
          (10 for best, 1 for worst)
        
        </label>
         
        <input
          class="form-control"
          id="satisfaction"
          max="10"
          min="1"
          placeholder="1-10"
          type="number"
        />
      </section>
       
      <section
        class="form-group"
      >
        <textarea
          class="form-control"
          id="message"
          placeholder="Describe your review..."
          rows="5"
        />
      </section>
       
      <button
        class="submitReview"
      >
        Submit!
      </button>
    </div>
     
    <!---->
  </div>
</div>
`;
